crates/ntx-builder/src/builder/mod.rs

```diff
@@ -94,6 +101,81 @@ pub struct NetworkTransactionBuilder {
 }
 
 impl NetworkTransactionBuilder {
+    pub async fn serve_new(self) -> anyhow::Result<()> {
+        // TODO: separate out the startup stuff so it can loop and repeat and wait on the network
+        // etc.
+        let store = StoreClient::new(&self.store_url);
+        let block_prod = BlockProducerClient::new(self.block_producer_address);
+
+        let unconsumed = store
+            .get_unconsumed_network_notes()
+            .await
+            .context("failed to fetch unconsumed notes from the store")?;
+        let (chain_tip, _mmr) = store
+            .get_current_blockchain_data(None)
+            .await
+            .context("failed to fetch the chain tip data from the store")?
+            .context("chain tip data was None")?;
+
+        let mut mempool_events = block_prod
+            .subscribe_to_mempool(chain_tip.block_num())
+            .await
+            .context("failed to subscribe to mempool events")?;
+
+        let mut state = crate::state::State::new(unconsumed.into_iter());
+        let mut interval = tokio::time::interval(self.ticker_interval);
```

Comment:
https://docs.rs/tokio/latest/tokio/time/enum.MissedTickBehavior.html#variant.Burst is the default behaviour if we miss a tick. Not sure that's intended.

Reply:
I think its.. probably fine. We should probably just `skip` instead

This does make me realize though what we definitely want to limit the overall number of in-progress tasks by skipping if `tasks.len() > N`. Thanks!